metrics/ci_integrations/lib/integration/interface/destination/client/destination_client.dart

```diff
@@ -18,4 +18,15 @@ abstract class DestinationClient extends IntegrationClient {
   ///
   /// Does nothing if a project with the given [projectId] is not found.
   Future<void> addBuilds(String projectId, List<BuildData> builds);
+
+  /// Fetches a list of builds with the given [status] for a project identified
+  /// by the given [projectId].
+  Future<List<BuildData>> fetchBuildsWithStatus(
+    String projectId,
+    BuildStatus status,
+  );
+
+  /// Updates the builds of the project identified by the given [projectId]
+  /// with the given [builds].
```

Comment:
```suggestion
  /// Updates the builds of the project identified by the given [projectId]
  /// with the given [builds].
  /// 
  /// Throws an [ArgumentError] if a project with the given [projectId]
  /// is not found.
```
Also, it would be great to update the `addBuilds` method documentation since it seems like it throws an `ArgumentError` too if the project is not found.